### ISSUE TITLE
Version Number? 

### ISSUE THREAD

**Issue by jiapei100** [halide/Halide#2776]
Hi, 
does Halide have a formal version number?

Cheers
Pei

**Comment by TomBobach** [comment]
I second that inquiry.

Halide team: thanks for answering!

**Comment by abadams** [comment]
No, we haven't gotten around to doing proper semantic versioning. Most major users of Halide just track trunk rather than using a released version, so version numbers haven't been anyone's top priority.

**Comment by abadams** [comment]
**Automated triage note:** this issue was flagged by Claude as likely safe to close, for the reason below. The reasoning was checked against the current source, and a second pass tried and failed to refute it, but no human verified this specific text. **If the reason is wrong, please reopen the issue and say so** — that is the expected outcome for some fraction of these, and no offence taken.

**Reason (CLOSE_FIXED):** The question was whether Halide has a formal version number. It now does: canonical semantic version macros live in the runtime header and the project ships tagged releases (v8 through v21, currently 22.0.0-dev).

**Evidence:** src/runtime/HalideRuntime.h:22-29 defines HALIDE_VERSION_MAJOR/MINOR/PATCH (22/0/0) and states this is the canonical version; git tags include v21.0.0, v20.0.0.dev0, v19.0.0.dev0, etc.

Closing: Halide has had semantic versioning and tagged releases for years — see HALIDE_VERSION_MAJOR/MINOR/PATCH in src/runtime/HalideRuntime.h and the release tags (currently v21 released, main is 22.0.0-dev).
